### PR TITLE
Set deterministic_policy of `exploration_wrapper` to False

### PR DESCRIPTION
**Pull request by kmdanielduan**
- Set deterministic_policy of `exploration_wrapper` to False.
- `exploration_wrapper.predict(...)` will
    - sample non-deterministic actions for stochastic algorithms.
    - generate deterministic actions for [deterministic algorithms](https://stable-baselines3.readthedocs.io/en/master/_modules/stable_baselines3/td3/policies.html) (e.g. TD3) because the flag of `deterministic_policy=False` is ignored when `.predict(obs)` is called.

Mentioned in #420 .

### CODE REVIEW

**Review by AdamGleave — COMMENTED** [review]
Might be worth adding `deterministic_policy=False` to the `generate_trajectories` calls? It defaults to `False` there, but might be worth reiterating that both to (a) reassure the casual reader; (b) be defensive in case the default changes.

Might also be worth adding a comment briefly explaining this choice, e.g. that the `buffering_wrapper` collects rollouts from a non-deterministic policy so we do that here as well for consistency.

File: nohup.out [hunk 1]
```diff
@@ -0,0 +1,18 @@
+
```
Comment (AdamGleave) [inline nohup.out:1]: I don't think you meant to commit this file?

File: src/imitation/algorithms/preference_comparisons.py [hunk 2]
```diff
@@ -161,13 +161,15 @@ def __init__(
         self.log_callback = self.venv.make_log_callback()
 
         self.algorithm.set_env(self.venv)
-        policy = rollout._policy_to_callable(
+        policy_callable = rollout._policy_to_callable(
             self.algorithm,
             self.venv,
-            deterministic_policy=True,
+            # we keep the default policy scheme in determinism by setting
```
Comment (AdamGleave) [inline src/imitation/algorithms/preference_comparisons.py:167]: Default where?

File: src/imitation/data/rollout.py [hunk 3]
```diff
@@ -282,6 +282,14 @@ def get_actions(states):
             return acts
 
     elif isinstance(policy, Callable):
+        # When a policy callable is passed, by default we will use it directly.
+        # We are not able to change the determinism of the policy when it is a
+        # callable that only takes in the states.
+        if deterministic_policy is True:
```
Comment (AdamGleave) [inline src/imitation/data/rollout.py:288]:
```suggestion
        if deterministic_policy:
```

File: src/imitation/data/rollout.py [hunk 4]
```diff
@@ -282,6 +282,14 @@ def get_actions(states):
...
+            raise ValueError(
```
Comment (AdamGleave) [inline src/imitation/data/rollout.py:289]: Good idea adding this check!

File: src/imitation/policies/exploration_wrapper.py [hunk 5]
```diff
@@ -30,21 +30,22 @@ def __init__(
         """Initializes the ExplorationWrapper.
 
         Args:
-            policy: The policy to randomize.
+            policy_callable: The policy callable to randomize. Preferably a policy
+                callable to match the same type of self._random_policy(obs) call.
```
Comment (AdamGleave) [inline src/imitation/policies/exploration_wrapper.py:34]: What does "match the same type" mean? It's type is a `rollout.PolicyCallable` (according to the annotation), what other type should be aligned?

**Review by AdamGleave — COMMENTED** [review]
Thanks, looks almost ready, main points:
  - nohup.out should be deleted
  - A few of the comments/docstring I found a bit hard to understand, can you edit where I had problems?

**Comment by kmdanielduan** [comment]
Sorry that I requested re-review by mistake. I'm on it now.

File: src/imitation/data/rollout.py (hunk 3 above)
Comment (kmdanielduan) [reply src/imitation/data/rollout.py:288]: Yeah!

File: src/imitation/policies/exploration_wrapper.py (hunk 5 above)
Comment (kmdanielduan) [reply src/imitation/policies/exploration_wrapper.py:34]: `def _random_policy(self, obs: np.ndarray) -> np.ndarray:` seems to be the same type as `PolicyCallable = Callable[[np.ndarray], np.ndarray]`. Perhaps no need to include it.

**Review by AdamGleave — APPROVED** [review]
LGTM. (Thanks for adding test coverage.)

**Pull request merged**